ipa-core/src/protocol/context/malicious.rs

```diff
@@ -208,18 +208,19 @@ impl<'a, F: ExtendableField> Upgraded<'a, F> {
             NotSharded,
         )
     }
-}
-
-#[async_trait]
-impl<'a, F: ExtendableField> UpgradedContext<F> for Upgraded<'a, F> {
-    type Share = MaliciousReplicated<F>;
 
-    fn share_known_value(&self, value: F) -> MaliciousReplicated<F> {
+    pub fn share_known_value(&self, value: F) -> MaliciousReplicated<F> {
         MaliciousReplicated::new(
             Replicated::share_known_value(&self.clone().base_context(), value),
             &self.inner.r_share * value.to_extended(),
         )
     }
+}
+
+#[async_trait]
+impl<'a, F: ExtendableField> UpgradedContext for Upgraded<'a, F> {
```

Comment:
When I originally introduced this `ExtendableField` thing to the codebase, it was to support malicious security for boolean circuits, by virtue of "extending" them with a GaloisField extension. That concept is now fully deprecated in our codebase, and we are instead relying on DZKPs for malicious security in boolean fields. This makes me wonder if we should just *eliminate* the whole `ExtendableField` concept from our codebase.

It used to denote: "There is some way in which elements of this field can be mapped to elements in a sufficiently large field, that there is enough security". I think for our remaining MAC based approach, there will be no mapping, we will just need to start with a sufficiently large field. If that's the case, consider just killing this thing entirely.